account_invoice_section_picking/models/account_move_line.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = "account.move.line"
+
+    def _get_section_group(self):
+        """Group invoice lines according to uninvoiced delivery pickings"""
+        group = super()._get_section_group()
+        # If product is invoiced by delivered quantities:
+        #  - filter on done pickings to avoid displaying backorders not yet
+        #    processed
+        #  - Remove pickings linked with same sale order lines if an invoice
+        #    was created after its date_done
+        invoice_section_grouping = self.company_id.invoice_section_grouping
+        if (
+            invoice_section_grouping == "delivery_picking"
+            and self.product_id.invoice_policy == "delivery"
+        ):
+            last_invoice_line = self.search(
+                [
+                    ("sale_line_ids", "in", self.sale_line_ids.ids),
+                    ("id", "not in", self.ids),
+                ],
+                order="create_date",
+                limit=1,
+            )
+            pickings_already_invoiced = self.env["stock.picking"].search(
+                [
+                    ("date_done", "<=", last_invoice_line.create_date),
```

Comment:
```suggestion
                    ("date_done", "<=", last_invoice_line.create_date),
```
this condition doesn't guarantee that this picking was invoiced, sale.order.line and fields like qty_to_invoice is more reliable data if we need concider only invoiced pickings

Reply:
@i-vyshnevska Do you see any use case where this code would not give the expected results?

Reply:
invoice can be canceled for some reason and a new one created, partial deliveries, partial invoices
create_date - is rather a technical field for the functional side there are other separate fields 
and having such difficulties to have a proper time in the test case is confirming that the functional side tries to rely on a side that shouldn't be here

Reply:
I added an extra domain condition to avoid selecting a canceled invoice. :+1: 
Then, partial deliveries are actually covered through this (it's the whole point of this override).
I agree that a reduction of the quantity on the invoice lines could bring issues, so I did state this in the readme as a known issue, because I don't see any possible way handle this use case properly.
Finally, I would have preferred to be able to avoid relying on `create_date` but we cannot rely on the `date` because it's only set on the posting. Then as the `qty_invoiced` is already taken into account when an invoice is in draft, the only other possibility to know what was delivered after the last invoice was generated would have been to define a generated date on the invoice, which would have been the same than the `create_date`. So I guess it's ok to use it in such a case and I also did state it in the readme.